### PR TITLE
add copy_cross_scope

### PR DESCRIPTION
**Pull request by Baibaifan**
### PR types

New features
### PR changes

OPs
### Describe

image

Add copy_cross_scope_op. This op is used by pipeline to copy tensors across micro batch scopes.  Copy the variable value of the giving Id's micro scope to the micro scope of Id + 1 position.  If need to copy back to the main scope, using to_main_scope option to copy the variable value of the current micro scope to the main scope.

### CODE REVIEW

File: paddle/fluid/operators/copy_cross_scope_op.cc [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      auto main_tensor = main_var->GetMutable<LoDTensor>();
+      TensorCopySync(*dst_tensor, main_tensor->place(), main_tensor);
+    }
+  }
+};
+
+class CopyCrossScopeOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  void Make() override {
+    AddInput("X", "Tensor to copy.");
```
Comment (chenwhql) [inline paddle/fluid/operators/copy_cross_scope_op.cc]: 这个输入输出的注释不太符合规范，建议写完整一些
image

File: paddle/fluid/operators/copy_cross_scope_op.cc [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    AddInput("Id", "ID of the current scope.");
+    AddAttr<bool>("ToM", "Return current scope to main scope.");
```
Comment (chenwhql) [inline paddle/fluid/operators/copy_cross_scope_op.cc]: Attribute一般都小写命名，而且tom也不太明确，要不写全`to_main_scope`？建议设置默认值，`SetDefault`

**Review by chenwhql — DISMISSED** [review]
LGTM for change in framework.py

**Review by chenwhql — APPROVED** [review]
LGTM for change in framework.py

**Review by phlrain — APPROVED** [review]

**Pull request merged**